src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -322,6 +327,61 @@ private void conjureSnapshotTablesFromAsset(AssetSpecification asset,
 
         snapshotSource.snapshotMapTables(mapTableList);
         snapshot.snapshotTables(tableList);
+
+        // now that the new snapshot tables are set, we map from the asset relationships into snapshot relationships
+        List<Relationship> sourceRelationships = asset.getAssetRelationships()
+            .stream()
+            .map(AssetRelationship::getDatasetRelationship)
+            .collect(Collectors.toList());
+        snapshot.relationships(createSnapshotRelationships(sourceRelationships, snapshot));
+    }
+
+    private Column getColumnByNameOrThrow(Table table, String columnName) {
+        Map<String, Column> columnLookup = table.getColumnsMap();
+        if (!columnLookup.containsKey(columnName)) {
+            String message = String.format("Cannot find column %s in table %s", columnName, table.getName());
+            throw new CorruptMetadataException(message);
+        }
+        return columnLookup.get(columnName);
+    }
+
+    /**
+     * Map from a list of source relationships (from a dataset or asset) into snapshot relationships. This does the
+     * mapping based on table names not being changed from the source to the destination currently since we do not
+     * expose the functionality to do this (even though the schema supports it).
```

Comment:
That seems unfortunate. The schema supports a way to do this without basing it on names, right?

Reply:
Definitely doable, and probably without extra queries. We need to keep track of a mapping from the old table id to the new one when we build it instead of relying on the name. Now that I type out that last sentence it doesn't seem like too much of a lift :)

Reply:
SnapshotMapTable and SnapshotMapColumn made this pretty easy -- should be set now.